packages/utils/src/utils/date.utils.ts

```diff
@@ -114,3 +114,12 @@ const createLabel = (labelKey: LabelKey, amount: bigint): LabelInfo => ({
  */
 export const nowInBigIntNanoSeconds = (): bigint =>
   BigInt(Date.now()) * BigInt(1e6);
+
+/**
+ * Converts a given `Date` object to a timestamp in nanoseconds as a `bigint`.
+ *
+ * @param {Date} date - The `Date` object to convert.
+ * @returns {bigint} The timestamp in nanoseconds.
+ */
+export const toBigIntNanoSeconds = (date: Date): bigint =>
+  BigInt(date.getTime()) * BigInt(1e6);
```

Comment:
Not a blocker, just for readability and to avoid unnecessary conversion. What do you think?

```suggestion
  BigInt(date.getTime()) * 1_000_000n;
```

Reply:
Good point, I just copied over the existing constant already used in `nowInBigIntNanoSeconds`.

I plan to provide a refactoring PR #851 to resolve duplicate code after this PR, I'll include your suggestion in there.